File: app/src/main/java/org/mesonet/app/advisories/BaseAdvisoryDataProvider.java
package org.mesonet.app.advisories;

import java.util.List;
import java.util.Observable;

import javax.inject.Inject;


public abstract class BaseAdvisoryDataProvider extends Observable
{
    protected List<AdvisoryModel> mCurrentAdvisories;

    @Inject
    AdvisoryParser mAdvisoryParser;



    public void SetData(String inData)
    {
        mCurrentAdvisories = mAdvisoryParser.ParseAdvisoryFile(inData);
        setChanged();
        notifyObservers();
    }



    public List<AdvisoryModel> GetAdvisories()
    {
        return mCurrentAdvisories;
    }



    public int GetCount()
    {
        if(mCurrentAdvisories == null)
            return 0;

        return mCurrentAdvisories.size();
    }


    protected abstract void Update();
}

File: app/src/main/java/org/mesonet/app/advisories/BaseAdvisoryDataProvider.kt
package org.mesonet.app.advisories

import java.util.Observable

import javax.inject.Inject


abstract class BaseAdvisoryDataProvider : Observable() {
    var mCurrentAdvisories: List<AdvisoryModel>? = null

    @Inject
    lateinit var mAdvisoryParser: AdvisoryParser


    fun SetData(inData: String) {
        mCurrentAdvisories = mAdvisoryParser.ParseAdvisoryFile(inData)
        setChanged()
        notifyObservers()
    }


    fun GetAdvisories(): List<AdvisoryModel>? {
        return mCurrentAdvisories
    }


    fun GetCount(): Int? {
        return if (mCurrentAdvisories == null) 0 else mCurrentAdvisories?.size

    }


    protected abstract fun Update()
}
